ccmlib/common.py

```diff
@@ -973,3 +974,17 @@ def print_if_standalone(*args, debug_callback=None, end='\n', **kwargs):
         print(*args, *kwargs, end=end)
     else:
         debug_callback(*args, **kwargs)
+
+
+def get_installed_scylla_package_hash(source_file: Path):
+    current_hash = ""
+
+    # If source file does not exists - we can not to check the hash of the existing package
+    if source_file.exists():
+        with open(source_file, 'r') as f:
+            lines = f.readlines()
+        # get hash from file
+        for line in lines:
+            if line.startswith("hash="):
+                current_hash = line.replace("hash=", "").strip()
```

Comment:
is it possible that the `source_file` does not contain a line of "hash=.."? is it possible that it contains multiple lines of "hash=.."? if both answers are "no", i'd suggest put something like:
```py
if not source_file.exists():
    return ""
with open(source_file, 'r') as f:
    for line in f:
        if line.startswith("hash="):
            return line.replace("hash=", "").strip()
throw ArgumentError(f"{source_file} does not contains the hash")
```

less indent this way. and less tolerant for sure.

Reply:
Few lines with "hash" is not possible because we replace whole file and not add the lines.

No "hash" it is possible. Due to this I added the comment:

https://github.com/scylladb/scylla-ccm/pull/557/files#:~:text=%23%20Current%20hash%20may,start%20downloading%20again

```
# Current hash may be None. It may be due to uncompleted downloading and installation.
            # Or because of it is old installation that was not saved the hash yet.
            # In any case the new download and installation should be performed.
            # For this goal we need to remove existing folder and start downloading again
```

Reply:
the source.txt file - version will be download

Reply:
i see. then still good to return early in that case.

Reply:
Why do you think we need to return and stop the test? Maybe there is the reason that I did not think?

Reply:
I think @tchaikov is talking more on the style, i.e. readability of it, nothing logical you are missing here. 
i.e. no need for exception, but the style suggestion still can be relevert

Reply:
@tchaikov I changed it, let me know if it is more clear

Reply:
thanks. indeed. i was concerned about the readability. the code was and still correct.